source/common/http/mixed_conn_pool.cc

```diff
@@ -15,6 +15,13 @@ Envoy::ConnectionPool::ActiveClientPtr HttpConnPoolImplMixed::instantiateActiveC
   if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.allow_concurrency_for_alpn_pool")) {
     // For now, use the hard-coded setting. Eventually use the cached setting if available.
     initial_streams = host()->cluster().http2Options().max_concurrent_streams().value();
+    if (http_server_properties_cache_ && origin_.has_value()) {
+      uint32_t cached_concurrency =
+          http_server_properties_cache_->getConcurrentStreams(origin_.value());
+      if (cached_concurrency != 0 && cached_concurrency < initial_streams) {
```

Comment:
nit: Maybe a comment here on why we don't use cached_concurrency if it's larger than initial. (Presumably it's because we don't want to be too aggressive?)

Reply:
done